bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala

```diff
@@ -44,6 +53,56 @@ class BitcoindRpcClient(val instance: BitcoindInstance)(
   require(version == BitcoindVersion.Unknown || version == instance.getVersion,
           s"bitcoind version must be $version, got ${instance.getVersion}")
 
+  // Chain Query Api
+
+  /** Gets the height of the given block */
+  override def getBlockHeight(
+      blockHash: DoubleSha256DigestBE): Future[Option[Int]] = {
+    getBlockHeader(blockHash).map(header => Some(header.height))
+  }
+
+  /** Gets number of confirmations for the given block hash */
+  override def getNumberOfConfirmations(
+      blockHash: DoubleSha256DigestBE): Future[Option[Int]] = {
+    getBlockHeader(blockHash).map(header => Some(header.confirmations))
+  }
+
+  /** Gets the number of compact filters in the database */
+  override def getFilterCount: Future[Int] =
```

Comment:
Not be implemented here and then implemented in the specific version types no?

Reply:
It is overridden for the versions that support it.

Reply:
It's bad idea to override in the super type. For instance, are you going to remember to update this for V20 & V21? You aren't going to get a compiler warning -- which makes it _super_ easy to forget.